cql3/restrictions/statement_restrictions.cc

```diff
@@ -1095,6 +1096,7 @@ bytes_opt get_kth(size_t k, const query_options& options, const ::shared_ptr<ter
 
 template<typename Range>
 value_list to_sorted_vector(const Range& r, const serialized_compare& comparator) {
+    BOOST_CONCEPT_ASSERT((boost::ForwardRangeConcept<Range>));
```

Comment:
Why are you using boost concepts? Isn't this std::ranges::forward_range?

Reply:
Because I'm passing boost ranges, which apparently don't satisfy the std requirement.  When I `require std::ranges::forward_range`, I get this error at a call site:
```
/usr/include/c++/10/concepts:67:28: note:   ‘std::forward_iterator_tag’ is not a base of ‘boost::iterators::detail::iterator_category_with_traversal<std::input_iterator_tag, boost::iterators::bidirectional_traversal_tag>’
```

Reply:
So how about using std::ranges instead of boost? I want to reduce use of new boost libraries, and I want our concepts to be real concepts, not boost macros.

Reply:
> So how about using std::ranges instead of boost?

That would be inconsistent with the other dozen uses within the same file.  I can look into replacing them all in a follow-up, of course.

Reply:
I don't want to add boost concepts. I'd like to restrict the amount of vocabulary a developer needs to learn in order to understand the code base, not to expand it.

The requirement to have consistency within the same source file makes it harder to make progress.

Reply:
Switched to using std ranges and concept.